gb-dbg/src/memory.rs

```diff
@@ -0,0 +1,67 @@
+use egui::Ui;
+use egui_memory_editor::option_data::MemoryEditorOptions;
+use egui_memory_editor::{MemoryEditor, ReadFunction, WriteFunction};
+use std::ops::Range;
+
+pub struct GBMemoryEditor<T> {
+    memory: T,
+    memory_editor: MemoryEditor<T>,
+}
+
+impl<T> GBMemoryEditor<T> {
+    pub fn draw(&mut self, ui: &mut Ui) {
+        self.memory_editor
+            .draw_editor_contents(ui, &mut self.memory);
+    }
+}
+
+pub struct MemoryEditorBuilder<'d, T> {
+    read_func: ReadFunction<T>,
+    write_func: Option<WriteFunction<T>>,
+    address_ranges: [(&'d str, Range<usize>); 3],
+    memory: T,
+}
+
+impl<T> MemoryEditorBuilder<'_, T> {
+    pub fn new(read_func: ReadFunction<T>, memory: T) -> Self {
+        let arr: [(&str, Range<usize>); 3] = [
+            ("WRam", 0..0xAA),
+            ("VRam", 0xAA..0xBB),
+            ("Placeholder", 0xBB..0xCC),
+        ];
```

Comment:
What these ranges represent, they're smaller than the one used by `address_bus` It's maybe because right now It's a POC ?

Reply:
:ok_hand: